spec/spec.md

```diff
@@ -231,89 +231,258 @@ Presentation Definitions are objects generate to articulate what proofs an entit
+:::
 
+If the `from` property contains an array of nested _Submission Requirement_
+objects, it directs the consumer of the _Presentation Definition_ to submit
+members such that each nested _Submission Requirement_ object is satisfied. In
+the following example, the `from` property contains an array of nested
+_Submission Requirement_ objects to require all members from groups `"A"`,
+`"B"`, and `"C"`:
+
+::: example Submission Requirement, all, nested
+  ```json
+  "submission_requirement": {
+    "name": "Picking all members from groups A, B, and C",
+    "purpose": "We need them all",
+    "rule": "all",
+    "from": [
+      {"rule": "all", "from": "A"},
+      {"rule": "all", "from": "B"},
+      {"rule": "all", "from": "C"},
+    ]
+  }
+  ```
+:::
 
 ##### `pick` rule
 
-Directs the consumer of the _Presentation Definition_ to submit a specified number of members from each of the matching `group` strings found in the _Requirement Object's_ `from` property. In the example below, the consuming entity would be required to submit one input from the `B` group and one input from the `C` group. Rule objects of the `pick` type are constructed as follows:
+- The _Submission Requirement_ object's `rule` property ****MUST**** contain
+  the string value `"pick"`.
+- The _Submission Requirement_ object ****MUST**** contain a `count` property,
+  and its value ****MUST**** be an integer greater than zero.
+
+If the `from` property contains a `group` string, it directs the consumer of
+the _Presentation Definition_ to submit all members of the matching `group`
+string. In the following example, the `from` property contains a `group`
+string to require a single member of group `"B"`:
+
+::: example Submission Requirement, pick, group
+  ```json
+  "submission_requirement": {
+    "name": "Picking one member from group B",
+    "purpose": "We only need one",
+    "rule": "pick",
+    "count": 1,
+    "from": "B"
+  }
+  ```
+:::
 
+If the `from` property contains an array of nested _Submission Requirement_
+objects, it directs the consumer of the _Presentation Definition_ to submit
+members such that the number of satisfied _Submission Requirement_ objects is
+exactly `count`. In the following example, the `from` property contains an
+array of nested _Submission Requirement_ objects to require either all members
+from group `"A"` or two members from group `"B"`:
+
+::: example Submission Requirement, pick, nested
   ```json
-  "rule": {
-    "type": "pick",
+  "submission_requirement": {
+    "name": "Either all from group A or two from group B",
+    "purpose": "Either way works",
+    "rule": "pick",
     "count": 1,
-    "from": ["B", "C"]
+    "from": [
+      {"rule": "all",  "from": "A"},
+      {"rule": "pick", "from": "B", "count": 2},
+    ]
+  }
```

Comment:
I like this use of nesting which allows a `pick` rule to operate as a logical `or`

Reply:
![image](https://user-images.githubusercontent.com/858687/84558168-c8472d80-acfe-11ea-855a-907da6718cf8.png)